src/rgw/rgw_cr_rados.h

```diff
@@ -1401,6 +1401,25 @@ class RGWRemoveObjCR : public RGWSimpleCoroutine {
   }
 };
 
+class LatencyMonitor {
+  ceph::mutex lock = ceph::make_mutex("LatencyMonitor::lock");
+  ceph::timespan total;
+  std::uint64_t count = 0;
+
+public:
+
+  LatencyMonitor() = default;
+  void add_latency(ceph::timespan latency) {
+    std::scoped_lock l(lock);
+    total += latency;
+    ++count;
+  }
+
+  ceph::timespan avg_latency() {
+    return total / count;
```

Comment:
to avoid divide-by-zero here, i guess this assumes we'll never call `avg_latency()` unless we've successfully acquired a lease?

Reply:
Oops. Done.